netlogo-gui/src/main/app/InterfaceTab.scala

```diff
@@ -2,12 +2,19 @@
 
 package org.nlogo.app
 
+import java.awt.event.{InputEvent, KeyEvent}
+
 import org.nlogo.core.I18N
-import org.nlogo.window.{WidgetInfo, EditDialogFactoryInterface, GUIWorkspace}
+import org.nlogo.window.{EditDialogFactoryInterface, GUIWorkspace, WidgetInfo}
 import org.nlogo.swing.Implicits.thunk2action
-import javax.swing.{JScrollPane, ScrollPaneConstants, Action, ImageIcon, AbstractAction, JSplitPane, JPanel}
-import java.awt.{Graphics2D, Graphics, Component, Container, ContainerOrderFocusTraversalPolicy, Dimension, BorderLayout}
+import javax.swing._
+import java.awt.{BorderLayout, Component, Container, ContainerOrderFocusTraversalPolicy, Dimension, Graphics, Graphics2D}
+
+import org.nlogo.editor.UndoManager
 
+object InterfaceTab {
+  var interfacePanelHadFocus = false
```

Comment:
I want to get rid of this static variable if at all possible. In a prior approach, you had an instance variable, which I would much prefer here.